tree/ntuple/v7/inc/ROOT/RNTupleUtil.hxx

```diff
@@ -172,6 +172,15 @@ struct RNTupleLocator {
    }
 };
 
+namespace Internal {
+template <typename T>
+auto MakeAliasedSharedPtr(T *rawPtr)
+{
+   static std::shared_ptr<T> fgRawPtrCtrlBlock;
+   return std::shared_ptr<T>(fgRawPtrCtrlBlock, rawPtr);
```

Comment:
Is that thread safe?

Reply:
a) There is no control block because `fgRawPtrCtrlBlock` is an empty `shared_ptr`.
b) If there was, reference counting is guaranteed to be thread-safe (usually implemented with atomics)

Reply:
> Initialization of function-local static variables is guaranteed to be thread-safe. 

Indeed but that's not the issue.

> Maybe less so the update of the reference count of the static shared_ptr.

The shared structure are thread safe but not the local objects (See for example https://medium.com/@yakupcengiz/thread-safety-with-std-shared-ptr-1597a7b4e93a).

But indeed that is not the point either since the above function can be written as:

```
   static const std::shared_ptr<T> fgRawPtrCtrlBlock;
   return std::shared_ptr<T>(fgRawPtrCtrlBlock, rawPtr);
```

So the function is thread safe but I am not sure that it does anything ...